XenModel/Actions/VM/ChangeMemorySettingsAction.cs

```diff
@@ -103,7 +92,22 @@ protected override void Run()
                     action = new VMCleanShutdown(VM);
                 else
                     action = new VMHardShutdown(VM);
-                action.RunExternal(Session);
+
+                try
+                {
+                    action.RunExternal(Session);
+                }
+                catch
+                {
+                    if (VM.power_state == vm_power_state.Halted || VM.current_operations.Any(op =>
+                        op.Value == vm_operations.clean_shutdown || op.Value == vm_operations.hard_shutdown ||
+                        op.Value == vm_operations.shutdown))
+                    {
+                        //ignore if it got already powered off or there are other shutting down tasks in progress

```

Comment:
If there are other shutting down tasks in progress but that haven't happened yet, will the code below (setting the memory) cope with this, does it need to wait until the shutdown has occurred (perhaps with rechecking, in case the other task is cancelled or fails)?

Reply:
It won't cope, it will reboot the VM with the same memory settings as before. I have to say I didn't manage to reproduce this scenario, but I can't exclude it might happen. Ok, Iet me revisit this.